### PR TITLE
Validation and snapshot for `adapter` and `template` configs

### PR DESCRIPTION
**Pull request by guptasu**
Add validation and snapshot loading of `adapter` and `template` configuration resources.

* validator.go validates the two kind of resources
* ephemeral.go snapshots the two kind of resources.
* Also, deleted the InfoRegistry. Not needed anymore as every Info comes in as a Custom resource and handled individually, we don't need any global registry.

NEXT PRs: load the handler and instance configs that refers to the adapter and template configs respectively.

### CODE REVIEW

File: mixer/pkg/runtime/config/descriptorUtil.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+		return nil, nil, "", fmt.Errorf("there has to be one proto file that has the extension %s", tmpl.E_TemplateVariety.Name)
+	}
+
+	var tmplName string
+	if nameExt, err := proto.GetExtension(tmplDescProto.GetOptions(), tmpl.E_TemplateName); err != nil {
+		return nil, nil, "", fmt.Errorf(
+			"proto files %s is missing required template_name option", tmplDescProto.GetName())
+	} else if err := validateTmplName(*(nameExt.(*string))); err != nil {
+		return nil, nil, "", err
+	} else {
```
Comment (geeknoid) [inline mixer/pkg/runtime/config/descriptorUtil.go]: No need for "else"

File: mixer/pkg/runtime/config/ephemeral.go [hunk 2]
```diff
@@ -234,31 +236,49 @@ func (e *Ephemeral) processInstanceConfigs(counters Counters) map[string]*Instan
...
-		log.Debugf("Processing incoming adapter info: name='%s'\n%s", adapterInfoKey.String(), cfg)
-		adapterInfos = append(adapterInfos, cfg)
-	}
 
-	log.Debugf("Total received adapter info: count=%d, value='%v'", len(adapterInfos), adapterInfos)
-	reg, err := newAdapterInfoRegistry(adapterInfos)
-	if err != nil {
-		log.Errorf("Error when reading adapter info='%v'", err)
-		counters.adapterInfoConfigError.Inc()
+		log.Debugf("Processing incoming adapter info: name='%s'\n%v", adapterName, cfg)
```
Comment (geeknoid) [inline mixer/pkg/runtime/config/ephemeral.go]: Should we introduce a logging scope for adapter config activity

File: mixer/pkg/runtime/config/descriptorUtil.go (hunk 1 above)
Comment (guptasu) [reply mixer/pkg/runtime/config/descriptorUtil.go]: D

File: mixer/pkg/runtime/config/ephemeral.go (hunk 2 above)
Comment (guptasu) [reply mixer/pkg/runtime/config/ephemeral.go]: You mean the zoning stuff ? Is that functionality in ? IMO, the entire config ingestion related errors can be in its own zone (or logging scope) and we don't need special zoning just for adapter config activities.

**Comment by guptasu** [comment]
Revoking this PR, since the branch is frozen and I am building more stuff on this PR. Will send an updated PR later.

**Pull request closed without merging**